lib/thrift/binary/framed/client.ex

```diff
@@ -81,14 +81,15 @@ defmodule Thrift.Binary.Framed.Client do
 
      - `send_timeout`: An integer that governs how long our connection waits when sending data.
 
-    `gen_server_opts`: A keyword list of options that control the gen_server behaviour.
+     - `gen_server_opts`: A keyword list of options that control the gen_server behaviour.
```

Comment:
I think this might be easier to use with the GenServer options as top level. For tcp the list is not a `Keyword.t` so we can't have those options at the top level, and must nest them. For GenServer start I think we could support `:name`, `:spawn_opt` and `:debug`. The other one is`:timeout` and that is not helpful to us because we don't do anything in `init/1` so never block, and then we don't need to worry about a key clash.

Also these options have different meaning to when `:gen_server_opts` is used elsewhere, which is confusing, and breaks the abstraction I think.

Reply:
By top level, do you mean being able to specify the options like
```elixir
Client.start_link("localhost", 8080, name: __MODULE__)
```
instead of
```elixir
Client.start_link("localhost", 8080, gen_server_opts: [name: __MODULE__])
```
Also I've noticed that I misinterpreted that line to have a missing `-`, when actually it was intentionally unindented as a separate list of options from `tcp_opts`.

Reply:
I did mean that yes but didn't notice this was just `-`. Lets go with this change :).